From 8163981ec3296c6768cacad2ac7382ae2c706853 Mon Sep 17 00:00:00 2001
From: Pavel Vomacka <pvomacka@redhat.com>
Date: Fri, 5 Aug 2016 14:04:03 +0200
Subject: [PATCH] Fix unicode characters in ca and domain adders

Topology graph didn't show plus icons correctly.

There is a problem with uglifying of javascript code. It does not leave unicode character
written in hexadecimal format unchanged. Therefore this workaround which inserts
needed character using Javascript function and uglifiyng does not affect it.

https://fedorahosted.org/freeipa/ticket/6175

Reviewed-By: Martin Basti <mbasti@redhat.com>
---
 install/ui/src/freeipa/topology_graph.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/install/ui/src/freeipa/topology_graph.js b/install/ui/src/freeipa/topology_graph.js
index ce2ebeaff611987ae27f2655b5da80bdcd1b4f8a..4bc3668647979c77719efa78b7a663d0e899216e 100644
--- a/install/ui/src/freeipa/topology_graph.js
+++ b/install/ui/src/freeipa/topology_graph.js
@@ -530,12 +530,14 @@ topology_graph.TopoGraph = declare([Evented], {
 
         function add_labels(type, color, adder_group) {
             var label_radius = 3;
+            var decimal_plus = parseInt('f067', 16); // Converts hexadecimal
+            // code of plus icon to decimal.
 
             var plus = adder_group
                 .append('text')
                 .classed('plus', true)
                 .classed(type + '_plus', true)
-                .text('\uf067');
+                .text(String.fromCharCode(decimal_plus));
 
             var label = adder_group.append('path')
                     .attr('id', type + '_label');
-- 
2.7.4

